notebooks/population_parquet_review.ipynb

```diff
@@ -0,0 +1,392 @@
+       "  <thead>\n",
+       "    <tr style=\"text-align: right;\">\n",
+       "      <th></th>\n",
+       "      <th>SUM</th>\n",
+       "      <th>MIN</th>\n",
+       "      <th>MAX</th>\n",
+       "      <th>MEAN</th>\n",
+       "      <th>id</th>\n",
+       "    </tr>\n",
+       "  </thead>\n",
+       "  <tbody>\n",
+       "    <tr>\n",
+       "      <th>16</th>\n",
+       "      <td>0.006808</td>\n",
+       "      <td>5.537143e-05</td>\n",
+       "      <td>0.000070</td>\n",
+       "      <td>0.000061</td>\n",
+       "      <td>860196107ffffff</td>\n",
+       "    </tr>\n",
+       "    <tr>\n",
+       "      <th>17</th>\n",
+       "      <td>0.004956</td>\n",
+       "      <td>2.208996e-06</td>\n",
+       "      <td>0.000045</td>\n",
+       "      <td>0.000034</td>\n",
+       "      <td>86019991fffffff</td>\n",
+       "    </tr>\n",
+       "    <tr>\n",
+       "      <th>26</th>\n",
+       "      <td>0.000330</td>\n",
+       "      <td>4.243886e-07</td>\n",
+       "      <td>0.000022</td>\n",
+       "      <td>0.000014</td>\n",
+       "      <td>8600a2817ffffff</td>\n",
+       "    </tr>\n",
+       "    <tr>\n",
+       "      <th>32</th>\n",
+       "      <td>0.008553</td>\n",
+       "      <td>3.608915e-05</td>\n",
+       "      <td>0.000058</td>\n",
+       "      <td>0.000041</td>\n",
+       "      <td>86015e7a7ffffff</td>\n",
+       "    </tr>\n",
+       "    <tr>\n",
+       "      <th>43</th>\n",
+       "      <td>0.003776</td>\n",
+       "      <td>1.950261e-06</td>\n",
+       "      <td>0.000048</td>\n",
+       "      <td>0.000041</td>\n",
+       "      <td>86015e827ffffff</td>\n",
+       "    </tr>\n",
+       "  </tbody>\n",
+       "</table>\n",
+       "</div>"
+      ],
+      "text/plain": [
+       "         SUM           MIN       MAX      MEAN               id\n",
+       "16  0.006808  5.537143e-05  0.000070  0.000061  860196107ffffff\n",
+       "17  0.004956  2.208996e-06  0.000045  0.000034  86019991fffffff\n",
+       "26  0.000330  4.243886e-07  0.000022  0.000014  8600a2817ffffff\n",
+       "32  0.008553  3.608915e-05  0.000058  0.000041  86015e7a7ffffff\n",
+       "43  0.003776  1.950261e-06  0.000048  0.000041  86015e827ffffff"
+      ]
+     },
+     "execution_count": 30,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "filtered_df = df[df['SUM'] > 0]\n",
+    "filtered_df.head()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 31,
+   "id": "6e838b53",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def hex_to_point(h):\n",
+    "    lat, lng = h3.h3_to_geo(h)\n",
+    "    return Point(lng, lat)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 35,
+   "id": "0523bf54",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "df['geometry'] = df['id'].apply(hex_to_point)\n",
+    "gdf = gpd.GeoDataFrame(df, geometry='geometry')"
+   ]
```

Comment:
~~In case this is a place where performance is important, you can do a vectorized conversion from h3 to shapely.~~

Well, in looking this up, it turns out that the latest h3-py releases removed the vectorized code that I added 3 years ago (although that apparently only had `geo_to_h3` and not `h3_to_geo`). h3-py development is such a mess IMHO. The core h3 v4 was released two years ago https://github.com/uber/h3/releases/tag/v4.0.0 but h3-py still hasn't had a v4 release.

In any case, you can use `shapely.points()` instead of `.apply(shapely.geometry.Point)` and that should make the shapely side much faster.

```
```

Also, h3 ids can be stored either as strings or as int64 or uint64. It is much, much more efficient to store h3 ids as int64/uint64, even more so in Parquet where it'll compress better.

Reply:
@kylebarron, thanks for the tips. It's great to know about storing the h3 IDs as int64 or uint64. 

Where do you see this fit in for the `shapely.points()`? I don't call `.apply(shapely.geometry.Point)`, but maybe I am missing something implicit that is obvious?

Reply:
Well the `shapely.geometry.Point` is internal to your `hex_to_point` wrapper. Ideally there would be a vectorized version of `h3.h3_to_geo` so we could avoid a Python for loop altogether, but it would still be a bit faster to do

```py
df[["latitude", "longitude"]] = df['id'].apply(h3.h3_to_geo)
geometry = shapely.points(df["longitude"], df["latitude"])
```